.github/workflows/build-stubs.yml

```diff
@@ -0,0 +1,165 @@
+            testbin: linux-test
+          - os: macos
+            runner: macos-latest
+            output: macos
+            testbin: macos-test
+          - os: windows
+            runner: windows-latest
+            output: windows.exe
+            testbin: windows-test.exe
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '~1.16.2'
+
+      - name: Compile x64 binary
+        id: compile
+        run: |
+          go build -o ${{ matrix.output }} stub.go
+          echo '' >> ${{ matrix.output }}
+          echo '### CAXA ###' >> ${{ matrix.output }}
+
+      - name: Test binary # Do a really simple smoke test of the stub binary
+        id: test
+        run: |
+          cp ${{ matrix.output }} ${{ matrix.testbin }}
+          echo 'success' > test.txt
+          tar -czf - test.txt >> ${{ matrix.testbin }}
+          printf "\n{ \"identifier\": \"${{ matrix.testbin }}/temp\", \"command\": [\"cat\", \"{{caxa}}/test.txt\"] }" >> ${{ matrix.testbin }}
+          RESULT=$(./${{ matrix.testbin }} 2>&1)
+          echo "RESULT=$RESULT" >> $GITHUB_ENV
+          [ "$RESULT" = 'success' ] || exit 1
+
+      - name: Test Results
+        if: failure()
+        run: |
+          echo "${{ matrix.os }} binary test failed:"
+          echo ${{ env.RESULT }}
+
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: stubs
+          path: "stubs/${{ matrix.output }}"
+
+  build_arm:
+    name: ${{ matrix.arch }} Binaries
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        platform: [linux/arm/v7, linux/arm64/v8]
+        include:
+          - platform: linux/arm/v7
+            arch: arm32v7
+            output: linux-armv7
+            testbin: linux-armv7-test
+          - platform: linux/arm64/v8
+            arch: arm64v8
+            output: linux-arm64
+            testbin: linux-arm64-test
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up QEMU
+        id: qemu
+        uses: docker/setup-qemu-action@v1
+        with:
+          platforms: ${{ matrix.platform }}
+
+      - name: Enable Docker Experimental Features
+        run: |
+          echo $'{\n  "experimental": true\n}' | sudo tee /etc/docker/daemon.json
+          mkdir -p ~/.docker
+          echo $'{\n  "experimental": "enabled"\n}' | sudo tee ~/.docker/config.json
+          sudo service docker restart
+          docker version -f '{{.Client.Experimental}}'
+          docker version -f '{{.Server.Experimental}}'
+  
+      - name: Build ${{ matrix.arch }}
+        run: |
+          docker run --rm --platform ${{ matrix.platform }} -v "$PWD":/usr/src/myapp -w /usr/src/myapp ${{ matrix.arch }}/golang:1.16 sh -c  "go build -o ${{ matrix.output }} stub.go && echo >> ${{ matrix.output }} && echo '### CAXA ###' >> ${{ matrix.output }}"
```

Comment:
Is this _actually_ producing ARM stubs? I see that QEMU is being _set up,_ but I don't see QEMU being _used_ to run an ARM Docker image, and cross compilation env vars like `GOARCH=arm GOARM=7` are not being used. If you can get hold of the assets (`linux-armv7`, `linux-arm64`), running `file` on them would produce some evidence, and the ultimate test would be to execute the assets on an actual ARM device (like the Raspberry Pi 3 (armv7) and Raspberry Pi 4 (arm64)).

My suggestion would be to drop QEMU and simply use Go's cross compilation on the ubuntu-20.04 image:
```
BIN=linux;       GOOS=linux   go build -o $BIN stub.go && echo >> $BIN && echo "### CAXA ###" >> $BIN
BIN=linux-armv6; GOOS=linux GOARCH=arm GOARM=6 go build -o $BIN stub.go && echo >> $BIN && echo "### CAXA ###" >> $BIN
BIN=linux-armv7; GOOS=linux GOARCH=arm GOARM=7 go build -o $BIN stub.go && echo >> $BIN && echo "### CAXA ###" >> $BIN
BIN=linux-arm64; GOOS=linux GOARCH=arm64       go build -o $BIN stub.go && echo >> $BIN && echo "### CAXA ###" >> $BIN
```

Reply:
Oh, now I see, `docker run --platform ${{ matrix.platform }} ${{ matrix.arch }}/golang:1.16` selects ARM images. If it works, that's fine. :-)

Reply:
Yes, the whole point of docker is to pull architecture specific build environments. I think using cross-compilation vs. emulation is a judgement call up to the maintainer. I could easily make an alternative workflow using cross-compilation and possibly include emulation just to test the binaries. However, if you want to avoid cross-compilation entirely, I think this is the best way to do it on actions.

There is a `file` output at the top post of this PR. I have a RasPi4 that I can test these on and get back to you.

Reply:
On a separate note, the linux binaries produced in this way are all dynamically linked. I wonder if it would be better to statically link them. It would only require setting `CGO_ENABLED=0` at build time.

Reply:
For `caxa`'s purposes and IoT scenarios, I think it would be better to produce statically linked binaries. 👍

Reply:
Agreed, see 2b6e113

Reply:
`file` output for statically linked binaries

```
linux:       ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, Go BuildID=ZpOuw6HzhSoP9EI3KQ_l/1v8uT7OxKkY5tzzfEQ22/oLODlNuykQXe-hrCuxwI/EEV3rKmMRZUXeD1IS7Uv, not stripped
linux-arm64: ELF 64-bit LSB executable, ARM aarch64, version 1 (SYSV), statically linked, Go BuildID=UyVMH9cEyLNPE1idNwee/oflnWHHVpTqfAYpPoa-i/I0BD9DzfYr7PjuG--Pso/Rdru9oR8nH_E46Qa_KrI, not stripped
linux-armv7: ELF 32-bit LSB executable, ARM, EABI5 version 1 (SYSV), statically linked, Go BuildID=p3j2M_NxOd4RaCq9opu2/LIjwcafgHr4oEqwc-Tym/nCKi9DrbUfyvPn_DwKUN/ZkXoH74Emni-t2CZuT60, not stripped
macos:       Mach-O 64-bit x86_64 executable
windows.exe: PE32+ executable (console) x86-64 (stripped to external PDB), for MS Windows
```

Reply:
> You can access all of the assets [here](https://github.com/maxb2/caxa/releases/tag/v1.0.0-test24) if you would like to test them too.

Thanks 👍   I have successfully tested the armv7 binary on a "real" Raspberry Pi 3.

Reply:
See fd5c12c and #4